### PR TITLE
ignores value delimiter if its between quotes

### PR DESCRIPTION
**Pull request by ganesh7**

### CODE REVIEW

File: lib/ical/parse.js [hunk 1]
```diff
@@ -66,6 +67,46 @@ ICAL.parse = (function() {
...
+    function getIndicesOf(searchStr, str, caseSensitive) {
+      var startIndex = 0, searchStrLen = searchStr.length;
+      var index, indices = [];
+      if (!caseSensitive) {
+          str = str.toLowerCase();
+          searchStr = searchStr.toLowerCase();
+      }
+      while ((index = str.indexOf(searchStr, startIndex)) > -1) {
+          indices.push(index);
+          startIndex = index + searchStrLen;
```
Comment (kewisch) [inline lib/ical/parse.js]: 2 space indent please

**Comment by lightsofapollo** [comment]
@ganesh7 this cuts the performance of parsing in half~... I want to fix this issue but that's not an acceptable slowdown.

**Comment by kewisch** [comment]
I haven't read the code in detail, but it looks like you are searching for all delimiters and then ignoring the value positon. Wouldn't it be better to parse the parameter value in a way that the quoted string is read and unquoted?

File: lib/ical/parse.js [hunk 2]
```diff
@@ -91,10 +91,20 @@ ICAL.parse = (function() {
       }
     }
```
Comment (lightsofapollo) [inline lib/ical/parse.js:91]: this works well but I think we need to restructure this function some more so we don't call valuePos.indexOf at the start... There are some assumptions about receiving values like this: "PARAM;X=true" (no value) which in practice never happens so we can probably do some more optimizations to bridge the final gap between the original time and the current time.

**Pull request closed without merging**